ng build fails with: An error occurred while extracting routes.

### Command

build

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

19.0.0-next.9

### Description

When running `ng build` on a freshly generated Angular 19 prerelease project, an error is thrown.

### Minimal Reproduction

- Run `npx @angular/cli@next new` and enable SSR
- Run `ng build`

### Exception or Error

```
✘ [ERROR] An error occurred while extracting routes.

Error: No handler function exported from /projects/ng19-test-proj/node_modules/@angular/build/src/utils/server-rendering/routes-extractor-worker.js
    at /projects/ng19-test-proj/node_modules/piscina/dist/worker.js:142:23
```

### Your Environment

Angular CLI: 19.0.0-next.12
Node: 20.18.0
Package Manager: npm 10.9.0
OS: darwin x64

Angular: 19.0.0-next.10
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1900.0-next.12
@angular-devkit/build-angular   19.0.0-next.12
@angular-devkit/core            19.0.0-next.12
@angular-devkit/schematics      19.0.0-next.12
@angular/cli                    19.0.0-next.12
@angular/ssr                    19.0.0-next.12
@schematics/angular             19.0.0-next.12
rxjs                            7.8.1
typescript                      5.6.3
zone.js                         0.15.0

### Anything else relevant?

_No response_

@alan-agius4 I copied the exact command you were using, still I'm getting the same error. Any chance that this is only happening on MacOS? Here is my full output:

```
Browser bundles      
Initial chunk files   | Names            |  Raw size | Estimated transfer size
main-B62CN3HO.js      | main             | 235.21 kB |                64.37 kB
polyfills-FFHMD2TL.js | polyfills        |  34.52 kB |                11.28 kB
styles-5INURTSO.css   | styles           |   0 bytes |                 0 bytes

                      | Initial total    | 269.73 kB |                75.65 kB


Server bundles       
Initial chunk files   | Names            |  Raw size
chunk-3LUFOK6H.mjs    | -                | 851.12 kB |                        
server.mjs            | server           | 842.21 kB |                        
polyfills.server.mjs  | polyfills.server | 268.60 kB |                        
main.server.mjs       | main.server      |  18.65 kB |                        
chunk-PES6N7ZL.mjs    | -                |   3.37 kB |                        

Lazy chunk files      | Names            |  Raw size
chunk-G752ZPIS.mjs    | xhr2             |  12.06 kB |                        
chunk-FS7GGNWD.mjs    | xhr2             |  12.06 kB |                        

Prerendered 0 static routes.
Application bundle generation failed. [25.262 seconds]

✘ [ERROR] An error occurred while extracting routes.

Error: No handler function exported from/projects/ssr-test/node_modules/@angular/build/src/utils/server-rendering/routes-extractor-worker.js
    at /projects/ssr-test/node_modules/piscina/dist/worker.js:142:23
```

--- a/packages/angular/build/src/tools/esbuild/javascript-transformer.ts
+++ b/packages/angular/build/src/tools/esbuild/javascript-transformer.ts
@@ -58,6 +58,8 @@ export class JavaScriptTransformer {
     this.#workerPool ??= new WorkerPool({
       filename: require.resolve('./javascript-transformer-worker'),
       maxThreads: this.maxThreads,
+      // Prevent passing `--import` (loader-hooks) from parent to child worker.
+      execArgv: [],
     });
 
     return this.#workerPool;

